test/framework/src/main/java/org/elasticsearch/test/index/IndexVersionUtils.java

```diff
@@ -92,6 +92,10 @@ public static IndexVersion getPreviousVersion(IndexVersion version) {
         return ALL_VERSIONS.get(place - 1);
     }
 
+    public static IndexVersion getPreviousMajorVersion(IndexVersion version) {
+        return IndexVersion.fromId(((version.id() / 1_000_000) - 1) * 1_000_000);
```

Comment:
Shouldn't this also add `99` like in https://github.com/thecoop/elasticsearch/blob/2af22faa54b526c2a0c8688e415f1bfbfa49f374/server/src/main/java/org/elasticsearch/Version.java#L310? So something like `7_14_00_99` becomes `8_00_00_99` -> `V_8_0_0`?

Reply:
Ah yes, but thats not going to hold for ever - it wont be the case for v9 onwards. I'll work round it

Reply:
Just for my knowledge/curiosity: do you know why 99 was added, up to now? Any historical reason?

Reply:
See the comments in `Version` - it used to represent a final release version, back when there were alpha & beta releases. That hasn't been the case for a long time, so '99' just represents a version that is released.